.changeset/chilly-ads-lay.md

```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/catalog-client': minor
```

Comment:
Let's hear the motivation for this not being a breaking change x)

Reply:
So we have a few alternatives:

- Roll back the client change entirely, for now
- Bump this to major to be strict about semver (in the hope that the cost of that actually not being significant)
- Leave this as minor, treating an _addition_ to a service API as nonbreaking (which is not strictly true, if people replace the API in question with their own implementations)

Reply:
Adding the `CatalogApi.getEntitiesByRef` method is breaking if you make your own custom catalog client. You'll also see that this PR addresses a lot of places in tests where they annoyingly used to have to change the way that they mock the catalog, just because of this API change, even though they weren't even calling 90% of the methods.

So there's for sure a chance that people will have to change their code in some places, after this.

Reply:
Yes. And we backed up and thought for a bit, what's the best way of actually dealing with this. Like, what's an actual breakage (when it comes to utility APIs) that's severe enough to warrant a major bump

Reply:
We could also reduce the impact of this kind of change by creating a testing helper that would consumers to extend an interface so that additions to the API would automatically update their types. What do you think?

Reply:
Yes, but would it be possible to make a generic one really, or would it end up with every single plugin needing to have a `my-plugin-test-utils` package just to be able to essentially expose jest types in there? The difficult thing is that you want to have some type of instance in your tests, where you can do e.g. `myInstance.someMethod.mockResolved...` etc, so you may want to make it part of the contract. Or some helper that essentially lets you do `mocked<ActualType>(myInstance).someMethod.mockResolved...` such that the `mocked` helper essentially accepts a `Partial<T extends ActualType>` or something like that and converts it to something that uses jest.Mocked perhaps? A bit unclear what the best DX here would be